src/graph/build.rs

```diff
@@ -18,21 +17,27 @@ use attachment::{Attachment, AttachmentImageViews, ColorAttachment, ColorAttachm
 use graph::Graph;
 use pass::{PassBuilder, PassNode};
 
-
+/// Color range for render targets
 pub const COLOR_RANGE: SubresourceRange = SubresourceRange {
     aspects: AspectFlags::COLOR,
     levels: 0..1,
     layers: 0..1,
 };
 
-
+/// Possible errors during graph building
 #[derive(Debug, Clone)]
 pub enum GraphBuildError<E> {
+    /// Framebuffer could not be created
     FramebufferError,
+    /// Shader module creation error
     ShaderError(ShaderError),
+    /// If no presentation render target is set
     PresentationAttachmentNotSet,
+    /// If no backbuffer is set
     BackbufferNotSet,
+    /// Allocation errors as returned by the `allocator` function given to `GraphBuilder::build`
     AllocationError(E),
+    /// Any other errors encountered during graph builder, such as graphics pipeline creation errors
```

Comment:
This is more like unknown error from backend. 
Pipeline cration error will cause panic as pass building code is expected to always configure it properly.